### PR TITLE
only allocate output array once in deserialization

### PR DESCRIPTION
**Pull request by gballet**

### CODE REVIEW

File: src/deserialize.zig [hunk 1]
```diff
@@ -37,10 +37,14 @@ inline fn safeReadSliceIntBig(comptime T: type, payload: []const u8, out: *T) vo
 
 // Returns the size of the payload as well as the offset to the
 // start of the actual data.
-fn sizeAndDataOffset(payload: []const u8) struct { size: usize, offset: usize } {
+fn sizeAndDataOffset(payload: []const u8) !struct { size: usize, offset: usize } {
```
Comment (gballet) [inline src/deserialize.zig:40]: I had to return an error here, which is actually a good thing: going over the encoded payload once requires more _a priori_ checks than before.

File: src/deserialize.zig [hunk 2]
```diff
@@ -105,25 +130,26 @@ pub fn deserialize(comptime T: type, serialized: []const u8, out: *T, allocator:
...
-                const r = sizeAndDataOffset(serialized);
+                const r = try sizeAndDataOffset(serialized);
                 var end = r.offset + r.size;
 
+                const list_size = try countRlpListItems(serialized[r.offset..end]);
+
                 // since it's not possible to say if the slice is "undefined", it has
                 // to be allocated regardless.
-                out.* = &[_]ptr.child{};
+                out.* = try allocator.alloc(ptr.child, list_size);
```
Comment (gballet) [inline src/deserialize.zig:148]: the allocation proper, this should be straightforward.

File: src/main.zig [hunk 3]
```diff
@@ -327,7 +327,7 @@ test "access list filled" {
 
     var buf: [128]u8 = undefined;
     const rlp = try std.fmt.hexToBytes(&buf, "f83af838f7940000000000000000000000000000000000001210e1a00000000000000000000000000000000000000000000000000000000000000203");
-    var out: StrippedTxn = StrippedTxn{ .access_list = &[0]AccessListItem{} };
+    var out: StrippedTxn = undefined; // StrippedTxn{ .access_list = &[0]AccessListItem{} };
```
Comment (gballet) [inline src/main.zig]: It no longer makes sense to partially allocate the structure, at least I can not think of a use case just now.

```suggestion
    var out: StrippedTxn = undefined;
```

Comment (jsign) [reply src/main.zig]: Cool.

**Review by jsign — APPROVED** [review]
Nice!

**Pull request merged**